src/main/java/org/terasology/biomesAPI/BiomeManager.java

```diff
@@ -128,11 +152,26 @@ public void checkBiomeChangeEvent(MovedEvent event, EntityRef entity) {
             if (!oldBiomeOptional.isPresent()) {
                 return;
             }
+
             Biome newBiome = newBiomeOptional.get();
             Biome oldBiome = oldBiomeOptional.get();
             if (oldBiome != newBiome) {
                 entity.send(new OnBiomeChangedEvent(oldBiome, newBiome));
+                metricsMode.setBiome(newBiome.getId());
             }
         }
     }
+
+    @ReceiveEvent(components = PlayerCharacterComponent.class)
+    public void checkPlayerSpawnedEvent(OnPlayerSpawnedEvent event, EntityRef entity) {
+        LocationComponent locationComponent = entity.getComponent(LocationComponent.class);
```

Comment:
If this method only makes sense if the location component is present you can just add it to argument list (that will automaticall ensure that it is not null). Otherwise, if done like this, a null check should be added before accessing `locationComponent`.

```suggestion
    public void checkPlayerSpawnedEvent(OnPlayerSpawnedEvent event, EntityRef entity, LocationComponent locationComponent) {
```